docs/source/json-api/index.rst

```diff
@@ -251,6 +253,102 @@ Nonempty Response with Unknown Template IDs Warning
         "status": 200
     }
 
+WebSocket ``/contracts/searchForever``
+======================================
+
+List currently active contracts that match a given query, with
+continuous updates.
+
+Two subprotocols must be passed, as described in `Choosing a party
+<#choosing-a-party>`__.
+
+application/json body must be sent first, formatted according to the
+:doc:`search-query-language`::
+
+    {"%templates": ["Iou:Iou"]}
+
+output a series of JSON documents, each ``argument`` formatted according
+to :doc:`lf-value-specification`::
+
+    {
+        "created": [{
+            "observers": [],
+            "agreementText": "",
+            "payload": {
+                "observers": [],
+                "issuer": "Alice",
+                "amount": "999.99",
+                "currency": "USD",
+                "owner": "Alice"
+            },
+            "signatories": ["Alice"],
+            "contractId": "#1:0",
+            "templateId": "f95486336ffb3c982319625bed0c88f68799b780b26b558b1e119277614ed634:Iou:Iou"
+        }]
+    }
+
+To keep the stream alive, you'll occasionally see messages like this,
+which can be safely ignored::
+
+    {"heartbeat": "ping"}
+
+After submitting an ``Iou_Split`` exercise, which creates two contracts
+and archives the one above, the same stream will eventually produce::
+
+    {
```

Comment:
This format differs from what you get from the `/commands/exercise` endpoint. Her you get something looking like
```
{
  created: [C1, C2],
  archived: [A1, A2]
}
```
There you'd instead get
```
[
  {created: C1},
  {created: C2},
  {archived: A1},
  {archived: A2}
]
```
I'd prefer if the encoding is the same for both endpoints. Since the `/commands/exercise` endpoint does not work with the first format, we need to use the second in both places. It would be nice if could keep the order of the events we don't filter out during consolidation for the `/contracts/searchForever` endpoint, but I don't consider it a hard requirement.